connections/http.go

```diff
@@ -50,12 +50,21 @@ func HTTPGetWithClient[T protoreflect.ProtoMessage](ctx context.Context, url str
 }
 
 func HTTPPostWithClient[T protoreflect.ProtoMessage](ctx context.Context, url string, client *http.Client, body interface{}, val T, authHeader string) error {
-	b, err := json.Marshal(body)
+	protoMsg := body.(proto.Message)
```

Comment:
why did you change the marshaler ? I vaguely remember we had issue with marshaling before. Don't remember details anymore. But this kinda change needs to be carefully tested before pushing to prod